src/codeflare_sdk/templates/base-template.yaml

```diff
@@ -157,23 +157,6 @@ spec:
                   - name: server-cert
                     mountPath: "/home/ray/workspace/tls"
                     readOnly: true
-                initContainers:
-                - command:
-                  - sh
-                  - -c
-                  - cd /home/ray/workspace/tls && openssl req -nodes -newkey rsa:2048 -keyout server.key -out server.csr -subj '/CN=ray-head' && printf "authorityKeyIdentifier=keyid,issuer\nbasicConstraints=CA:FALSE\nsubjectAltName = @alt_names\n[alt_names]\nDNS.1 = 127.0.0.1\nDNS.2 = localhost\nDNS.3 = ${FQ_RAY_IP}\nDNS.4 = $(awk 'END{print $1}' /etc/hosts)\nDNS.5 = rayclient-deployment-name-$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).server-name">./domain.ext && cp /home/ray/workspace/ca/* . && openssl x509 -req -CA ca.crt -CAkey ca.key -in server.csr -out server.crt -days 365 -CAcreateserial -extfile domain.ext
-                  image: rayproject/ray:2.7.0
-                  name: create-cert
-                  # securityContext:
-                  #   runAsUser: 1000
-                  #   runAsGroup: 1000
-                  volumeMounts:
-                  - name: ca-vol
-                    mountPath: "/home/ray/workspace/ca"
-                    readOnly: true
-                  - name: server-cert
-                    mountPath: "/home/ray/workspace/tls"
-                    readOnly: false
```

Comment:
@Fiona-Waters This initContainer should be kept because it's responsible for enabling the SSL communication on the RayClient. Enabling SSL is a requirement for  exposing the RayClient via the rayclient route which is used for enabling the local_interactive development.
The local_interactive allows you to directly connect from you laptop to the rayclient inside the k8s cluster.

Reply:
Thank you for the feedback. I have updated to keep the required initContainers and have created a separate template for comparison in the related unit test.